.azure-pipelines/client.yml

```diff
@@ -38,18 +33,42 @@ jobs:
       twine check $(Build.ArtifactStagingDirectory)/*
     displayName: 'Verify Readme'
 
-  - task: CopyFiles@2
-    displayName: 'Move Dependencies Report to Build Artifacts'
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifacts'
+    condition: succeededOrFailed()
+
+- job: 'Analyze'
+  dependsOn:
+    - 'Build_Target_Packages'  
+  pool:
+    vmImage: 'ubuntu-16.04'     
+  steps:  
+  - task: UsePythonVersion@0
+    displayName: 'Use Python $(python.version)'
+    inputs:
+      versionSpec: '$(python.version)'
+
+  - script: |
+     pip install wheel Jinja2
+    displayName: 'Prep Environment'
+    
+  - task: PythonScript@0
+    displayName: 'Analyze dependencies'
+    enabled: true
     inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: dependencies.html
-      TargetFolder: '$(Build.ArtifactStagingDirectory)'
-    condition: always()
+      scriptPath: 'scripts/analyze_deps.py'
+      arguments: '--verbose --out "$(Build.ArtifactStagingDirectory)/dependencies.html"'
+
+  - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+    # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
+    # builds should be sufficient.
+    condition: and(always(), ne(variables['System.TeamProject'], 'public'))
+    displayName: 'Component Detection'
 
   - task: PublishBuildArtifacts@1
```

Comment:
I think `PublishBuildArtifacts` should be removed from the `Analyze` job.  It should only be needed in the `Build` job.

Reply:
Actually the task 'Analyze Dependencies' is generating a file dependencies.html which we want to make a part of the artifacts. So effectively, this particular Publish Artifact step is actually publishing that file from the artifacts staging directory

Reply:
Please add a comment explaining this

Reply:
@bsiegel should we consider putting the dependency output into an different artifact zip file?

Reply:
So far, I have just been using directories under the default "drop" artifact, for example in Java we now have :

```
drop/
  java-8/
  java-lts/
  reports/
```

We could potentially have multiple named artifacts instead. I don't have a strong opinion but we would have to change a number of release pipelines

Reply:
Oh no is that what the java artifacts look like now? The release pipelines currently try to publish everything in the artifacts. We should at least consider having a "packages" or "release" artifact that will only contain what we need to release to our package managers.